src/modules/tas_studio/editor/mod.rs

```diff
@@ -4057,6 +4287,33 @@ fn replace_multiple_params<'a>(
     Some((first_line_idx, count, to))
 }
 
+fn enable_mouse_look() {
+    let marker = unsafe { MainThreadMarker::new() };
+    sdl::set_relative_mouse_mode(marker, true);
+    client::activate_mouse(marker, true);
+}
+
+fn disable_mouse_look() {
+    let marker = unsafe { MainThreadMarker::new() };
+    sdl::set_relative_mouse_mode(marker, false);
+    client::activate_mouse(marker, false);
+}
+
+fn get_viewangles() -> [f32; 3] {
+    let marker = unsafe { MainThreadMarker::new() };
+    let mut view_angles = [0.; 3];
+
+    unsafe { engine::hudGetViewAngles.get(marker)(&mut view_angles) };
+
+    view_angles
+}
+
+fn change_origin(origin: Vec3) {
+    let marker = unsafe { MainThreadMarker::new() };
+
+    unsafe { bxt::BXT_TAS_STUDIO_FREECAM_SET_ORIGIN.get(marker)(origin.into()) };
+}
+
```

Comment:
So, the editor is supposed to be completely independent from the surrounding HL environment. Basically, you should be able to put all Editor code into any other environment and pass the right variables, and it will work.

So for these callbacks, I guess the easiest way is to put them in a struct like
```
struct Callbacks {
    enable_mouse_look: fn(),
    // ...
}
```
and pass this struct into `Editor::tick()`.